### PR TITLE
boards: arm: Introduce dragonclaw board

### PR DESCRIPTION
**Pull request by duda-patryk**
Dragonclaw is a board created by Google for fingerprint-related functionality development. Board schematics, layout and BOM is available at:
https://chromium.googlesource.com/chromiumos/platform/ec/+/HEAD/docs/schematics/dragonclaw

Best regards,
Patryk

### CODE REVIEW

**Comment by semihalf-duda-patryk** [comment]
Fixed coding guidelines issue.

**Review by erwango — CHANGES_REQUESTED** [review]
Provided files are good, but we miss a proper board documentation, which would explain how to flash and debug for instance.

File: boards/arm/dragonclaw/Kconfig.board [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2022 Google Inc
```
Comment (fabiobaltieri) [inline boards/arm/dragonclaw/Kconfig.board:1]: should be "Google LLC" ("Copyright (c) 2022 Google LLC")

File: boards/arm/dragonclaw/Kconfig.board [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+# Dragonclaw is a board created by Google for fingerprint-related functionality
+# development. Board schematics, layout and BOM is available at:
+# https://chromium.googlesource.com/chromiumos/platform/ec/+/HEAD/docs/schematics/dragonclaw
```
Comment (fabiobaltieri) [inline boards/arm/dragonclaw/Kconfig.board:6]: This should go in the board documentation file (see Erwan comment)

File: boards/arm/dragonclaw/Kconfig.defconfig [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2022 Google Inc
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_DRAGONCLAW
+
+config BOARD
+	default "dragonclaw"
```
Comment (fabiobaltieri) [inline boards/arm/dragonclaw/Kconfig.defconfig:7]: Should we get into the habit of prefixing codenamed boards with `google_`? For example we currently have `google_kukui` already, would make it easy to spot them and give us a change to have a common maintainer or codeowner entry if needed. @keith-zephyr for an extra opinion

File: boards/arm/dragonclaw/dragonclaw.dts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+ */
+
+/dts-v1/;
+
+#include <st/f4/stm32f412Xg.dtsi>
+#include <st/f4/stm32f412z(e-g)tx-pinctrl.dtsi>
+
+/ {
+	model = "Google Dragonclaw development board";
+	compatible = "google,dragonclaw-fpmcu", "st,stm32f412";
```
Comment (fabiobaltieri) [inline boards/arm/dragonclaw/dragonclaw.dts:14]: I think this should just be `compatible = "google,dragonclaw-fpmcu"`, not that it's used for anything, but it should refer to the specific board afaik.

File: boards/arm/dragonclaw/dragonclaw.dts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+		zephyr,console = &usart2;
+		zephyr,shell-uart = &usart2;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,flash-controller = &flash;
+	};
+};
+
+&clk_hsi {
+	/* HSI clock frequency is 16Mhz */
```
Comment (fabiobaltieri) [inline boards/arm/dragonclaw/dragonclaw.dts:26]: nit: MHz (more below)

File: boards/arm/dragonclaw/dragonclaw.dts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+	status = "okay";
+};
+
+&clk_lsi {
+	/* LSI clock frequency is 32768khz */
```
Comment (fabiobaltieri) [inline boards/arm/dragonclaw/dragonclaw.dts:31]: nit: kHz

File: boards/arm/dragonclaw/dragonclaw.dts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+&rcc {
+	clocks = <&pll>; /* Select PLL as SYSCLK source (96Mhz) */
+	ahb-prescaler = <1>; /* SYSCLK not divided */
+	clock-frequency = <DT_FREQ_M(96)>; /* AHB frequency */
+	apb1-prescaler = <2>; /* AHB clock divided by 2 */
+	apb2-prescaler = <2>; /* AHB clock divided by 2 */
+};
+
+/* USART1: PA9/PA10 (TX/RX) to AP (Host Commands and MKBP) */
```
Comment (fabiobaltieri) [inline boards/arm/dragonclaw/dragonclaw.dts:52]: More of a personal preference, I would drop the `USART1: PA9/PA10 (TX/RX)` and just write something like `AP UART (Host Commands and MKBP)`. Reasoning is: the information is immediately available in the lines below and that's the actual source of truth for the configuration, duplicating it in the comment adds one more place to keep in sync manually, may end up copypasted wrongly in a future board variant and make the whole thing confusing. Same for the other node comments.

File: boards/arm/dragonclaw/dragonclaw.yaml [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+toolchain:
+  - zephyr
+  - gnuarmemb
+  - xtools
+ram: 256
+flash: 1024
+testing:
+  ignore_tags:
+    - net
+    - bluetooth
```
Comment (fabiobaltieri) [inline boards/arm/dragonclaw/dragonclaw.yaml:14]: I don't think these are needed, there's not ble or network device in the dts that could trigger these tests anyway.

File: boards/arm/dragonclaw/Kconfig.board (hunk 1 above)
Comment (semihalf-duda-patryk) [reply boards/arm/dragonclaw/Kconfig.board]: Done

File: boards/arm/dragonclaw/Kconfig.board (hunk 2 above)
Comment (semihalf-duda-patryk) [reply boards/arm/dragonclaw/Kconfig.board]: Fixed. Thanks!

File: boards/arm/dragonclaw/Kconfig.defconfig (hunk 3 above)
Comment (semihalf-duda-patryk) [reply boards/arm/dragonclaw/Kconfig.defconfig:7]: Let's wait for Keith's opinion.

File: boards/arm/dragonclaw/dragonclaw.dts (hunk 6 above)
Comment (semihalf-duda-patryk) [reply boards/arm/dragonclaw/dragonclaw.dts]: Done.

File: boards/arm/dragonclaw/dragonclaw.dts (hunk 5 above)
Comment (semihalf-duda-patryk) [reply boards/arm/dragonclaw/dragonclaw.dts]: Done.

**Comment by semihalf-duda-patryk** [comment]
Added documentation, fixed copyright, typos and style.

**Comment by semihalf-duda-patryk** [comment]
Added `board.cmake` and updated docs as @fabiobaltieri suggested. Confirmed that `west flash -r jlink` works.

File: boards/arm/dragonclaw/doc/index.rst [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+- USART_1 TX/RX : PA9/PA10
+- USART_2 TX/RX : PA2/PA3
+- SPI_1 CS/CLK/MISO/MOSI : PA4/PA5/PA6/PA7
+- SPI_2 CS/CLK/MISO/MOSI : PB12/PB13/PB14/PB15
+
+Programming and Debugging
+*************************
+
+Build application as usual for the ``dragonclaw`` board, and flash
+using μServo or SWD. If μServo is used, please follow the `Chromium EC Flashing
```
Comment (fabiobaltieri) [inline boards/arm/dragonclaw/doc/index.rst:33]: Instead of SWD you could say something like "using an external J-Link connected to J4.

Comment (semihalf-duda-patryk) [reply boards/arm/dragonclaw/doc/index.rst]: Done

File: boards/arm/dragonclaw/Kconfig.defconfig (hunk 3 above)
Comment (keith-zephyr) [reply boards/arm/dragonclaw/Kconfig.defconfig:7]: I agree with Fabio. Let's prefix boards with `google_`.

**Review by keith-zephyr — CHANGES_REQUESTED** [review]

File: boards/arm/dragonclaw/Kconfig.defconfig (hunk 3 above)
Comment (semihalf-duda-patryk) [reply boards/arm/dragonclaw/Kconfig.defconfig]: Done! Thanks!

**Comment by semihalf-duda-patryk** [comment]
Renamed board from `dragonclaw` to `google_dragonclaw`.

**Review by fabiobaltieri — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

**Pull request merged**